airflow/contrib/operators/gcf_function_deploy_operator.py

```diff
@@ -0,0 +1,202 @@
+    dict(name="description", regexp="^.+$", optional=True),
+    dict(name="entryPoint", optional=True, regexp=r'^\w+$'),
+    dict(name="runtime", regexp=r'^nodejs6$|^nodejs8$|^python37$', optional=True),
+    dict(name="timeout", regexp=r'^\d+(\.\d{1,9})?s$', optional=True),
+    dict(name="availableMemoryMb", custom_validation=_validate_available_memory_in_mb, optional=True),
+    dict(name="labels", type="dict", optional=True),  # TODO: add validation
+    # key: [\\p{Ll}\\p{Lo}][\\p{Ll}\\p{Lo}\\p{N}_-]{0,62}
+    # value: [\\p{Ll}\\p{Lo}\\p{N}_-]{0,63}
+    dict(name="environmentVariables", type="dict", optional=True),  # TODO(jarek) add validation:
+    # key: no empty string, null character, or equal sign: "(s?)[^\\x00=]+"
+    # value: "(s?)[^\\x00]*"
+    dict(name="network", regexp=r'^(projects/[^/]+/global/networks/(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?))|'
+                                r'((?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?))$',
+         optional=True),
+    dict(name="maxInstances", optional=True, custom_validation=_validate_max_instances),
+
+    dict(name="source_code", type="union", fields=[
+        dict(name="sourceArchiveUrl", regexp=r'^gs://[^\s]+$'),
+        dict(name="sourceRepositoryUrl", regexp=r'^gs://[^\s]+$', api='v1beta2'),
+        dict(name="sourceRepository", type="dict", fields=[
+            dict(name="url", regexp=r'(^https://source.developers.google.com/projects/'
+                                    r'[^/]+/repos/[^/]+/revisions/[^/]+(/paths/[^/]+)?$)|'
+                                    r'(^https://source.developers.google.com/projects/'
+                                    r'[^/]+/repos/[^/]+/moveable-aliases/[^/]+(/paths/[^/]+)?$)|'
+                                    r'(^https://source.developers.google.com/projects/'
+                                    r'[^/]+/repos/[^/]+/fixed-aliases/[^/]+(/paths/[^/]+)?$)')
+        ]),
+        dict(name="sourceUploadUrl", regexp=r'^https://[^\s]+$')  # TODO(jarek): should be secure ?
+    ]),
+
+    dict(name="trigger", type="union", fields=[
+        dict(name="httpsTrigger", type="dict"),
+        dict(name="eventTrigger", type="dict", fields=[
+            dict(name="eventType", regexp=r'^providers/[^/]+/eventTypes/[^/]+.[^/]+$'),
+            dict(name="resource", regexp=r'^[^/]$'),
+            dict(name="eventType", regexp=r'providers/[^/]+/eventTypes/[^/]+.[^/]+'),
+            dict(name="failurePolicy", type="dict", fields=[
+                dict(name="retry", type="dict")
+            ])
+        ])
+    ]),
+]
+
+
+class GCFFunctionDeployOperator(BaseOperator):
+    """
+    Create a function in Google Cloud Functions.
+
+    :param project_id: Project ID that the operator works on
+    :type project_id: str
+    :param region: Region where the operator operates on
+    :type region: str
+    :param body: Body of the cloud function definition
+    :type body: dict or google.cloud.functions.v1.CloudFunction
```

Comment:
Is there GCP documentation you can link to for body?

Reply:
Yep, We will, it's a good idea.